src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -402,14 +405,15 @@ private TypeSpec CreateDictionarySpec(TypeParseInfo typeParseInfo, ITypeSymbol k
                     {
                         populationCastType = CollectionPopulationCastType.NotApplicable;
                     }
-                    else if (_typeSymbols.GenericIDictionary is not null && GetInterface(type, _typeSymbols.GenericIDictionary_Unbound) is not null)
+                    else if (isDictionary)
                     {
                         // implements IDictionary<,> -- cast to it.
                         populationCastType = CollectionPopulationCastType.IDictionary;
                     }
                     else
                     {
-                        return CreateUnsupportedCollectionSpec(typeParseInfo);
+                        // not a dictionary
```

Comment:
Can't this type is implementing a non-generic IDictionary?

Reply:
It could, but we don't treat those as bindable dictionaries.
https://github.com/dotnet/runtime/blob/2106bb0c57c7bc39c9b5320668c09321651b99e6/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L447-L452

I'm not thrilled with the naming of this local, but it was the best I could come up with.  Here the "not a dictionary" means we're going to treat the return value as not a dictionary so we should fallback to object binding behavior.